schema/workflow.json

```diff
@@ -1672,47 +1683,45 @@
     "statedatafilter": {
       "type": "object",
       "properties": {
-        "dataInputPath": {
+        "input": {
           "type": "string",
-          "description": "JsonPath definition that selects parts of the states data input"
+          "description": "Workflow expression to filter the states data input"
         },
-        "dataOutputPath": {
+        "output": {
           "type": "string",
-          "description": "JsonPath definition that selects parts of the states data output"
+          "description": "Workflow expression that filters the states data output"
         }
       },
       "required": []
     },
     "eventdatafilter": {
       "type": "object",
       "properties": {
-        "dataOutputPath": {
+        "data": {
           "type": "string",
-          "description": "JsonPath definition that selects parts of the event data, to be merged with the states data"
+          "description": "Workflow expression that filters of the event data (payload)"
+        },
+        "toStateData": {
+          "type": "string",
+          "description": " Workflow expression that selects a state data element to which the event payload should be added/merged into. If not specified denotes the top-level state data element."
```

Comment:
could change to "selection expression" . would that be clearer?

Reply:
That could work, I'd be interested to hear others thoughts if they have any.  

I think the really confusing part is that the examples all have '${ .data }' which to me implies something to be extracted from the stateData, rather than indicating a location in the state-data where the results should be merged.

Reply:
Do you mean the "data" element? This comes from the CE format where the event payload is in the "data" context attribute. and most of the time "data" is the top-level property

Reply:
I'm sorry, my question isn't very clear.

```
   "toStateData": ".foo"
```
This looks like exactly what it is, something indicating a location in a json structure where `results` should be merged into the state data.

```
   "toStateData": "${ .foo }"
```
This is completely unclear, to the uninitiated, what this is supposed to do.  Is it extracting some value from .foo, and that value will tell the runtime where to merge the `results`?  Or, is the runtime supposed to use '.foo' as the location where the merge should occur?

The `${}` indicates to the reader than some interpolation or evaluation of that expression is to occur, and the results of that expression are what should actually be used so expressing toStateData in this way is extremely confusing.

Reply:
i do think i see your point. each expression has a result so toStateData might not fit maybe? 
the problem i have is then if its not an expression, but the way you have it ".foo" is still an expression ....
the intent is to say "merge into the "foo" element of the state data and create one if it does not exist"

I think we should create an issue and bring it up in a meeting to see opinions but I think for this pr its fine